react/features/authentication/middleware.ts

```diff
@@ -305,3 +279,27 @@ function _handleLogin({ dispatch, getState }: IStore) {
 
     dispatch(openTokenAuthUrl(tokenAuthServiceUrl));
 }
+
+/**
+ * Handles logout challenge. Opens logout dialog and hangs up the conference.
+ *
+ * @param {Store} store - The redux store in which the specified {@code action}
+ * is being dispatched.
+ * @param {string} logoutUrl - The url for logging out.
+ * @returns {void}
+ */
+function _handleLogout({ dispatch, getState }: IStore) {
+    const state = getState();
+    const { conference } = state['features/base/conference'];
+
+    const config = state['features/base/config'];
+    const logoutUrl = config.tokenLogoutUrl;
+
+    if (!conference) {
+        return;
+    }
+
+    if (logoutUrl) {
```

Comment:
This changes the behavior we had, now logout works only with logout URL. 
If you have configured your deployment with authentication with username and password and anonymous domain those that are authenticated can do a logout which sends an iq to jicofo and then hangups redirecting to welcomepage.